src/express-tooling.ts

```diff
@@ -105,27 +114,39 @@ function handleOAuthRequestMiddleware(options: MiddlewareOptions): RequestHandle
     const authHeader = getHeaderValue(req, AUTHORIZATION_HEADER_FIELD_NAME);
 
     if (!authHeader) {
+      if (logger) {
```

Comment:
I dont't like having the `if (logger)` code blocks all over. What do you think about creating a function around?

Reply:
You mean a type guard for the logger? Might work - even thought about that yesterday - I also don't like the NULL check distributed through the code. @bzums and I decided to distribute it in the first step, as just 2 logger calls were within the code. With additional logging it looked weird throughout the code. But I already got an idea how to solve this

Reply:
One further idea: we could also save a lot of if statements if we would move the logging to `rejectRequest`, we could extend its signature to something like:
```ts
const rejectRequest = (res: Response, { error: string, status?: number }, logger?: Logger): void
```

What do you think @ISO50 @Retro64 ?

Reply:
Thought about it, but I would prefer a more generalized approach for the logger - see next commit

Reply:
@bzums not so wrong idea, as it was the initial intention to make the result of the response visible in the logs, right?
Optional idea would be that the user could overwrite the reject method, this would allow the user to customize the reject, for example other error codes / messages in the body / return problem json.
Plus the user could easily determine log level and the format of the log message on it's own.
What do you think @bzums @Retro64 ?